plugin/manager.go

```diff
@@ -0,0 +1,44 @@
+package plugin
+
+import (
+	"fmt"
+	goplugin "plugin"
+)
+
+// Manager is in charge of instanciating plugins and storing those in memory
+type Manager struct {
+	middlewares []*Middleware
```

Comment:
Does this have to be a slice of pointers, or could it be simplified to a slice of values?